spring-cloud-stream-docs/src/main/asciidoc/spring-cloud-stream-overview.adoc

```diff
@@ -81,147 +71,586 @@ public class StreamApplicationTests {
+public class TransformProcessor {
+	@Transformer(inputChannel = Processor.INPUT, outputChannel = Processor.OUTPUT)
+	public Object transform(String message) {
+		return message.toUpper();
+	}
+}
+----
 
-=== Binder Selection
+===== @StreamListener for automatic content type handling
+
+Complementary to the Spring Integration support, Spring Cloud Stream provides a
+`@StreamListener` annotation of its own modeled by the other similar Spring Messaging
+annotations (e.g. `@MessageMapping`, `@JmsListener`, `@RabbitListener`, etc.)
+especially for dealing with use cases that involve content type management
+and type coercion.
+
+This approach is suited for applications that process inbound data that specifies content types, as Spring Cloud Stream provides an extensible `MessageConverter` registration mechanism for handling data conversion by the bound channels and for dispatching to `@StreamListener` annotated methods.
+
+For example, an application that processes external `Vote` events can be declared as follows:
+
+[source,java]
+----
+@EnableBinding(Sink.class)
+public class VoteHandler {
+
+  @Autowired
+  VotingService votingService;
+
+  @StreamListener(Sink.INPUT)
+	public void handle(Vote vote) {
+		votingService.record(vote);
+	}
+}
+----
+
+The fundamental distinction between this approach and a Spring Integration `@ServiceActivator` becomes relevant if one considers an inbound `Message` with a `String` payload and a `contentType` header of `application/json`. For `@StreamListener`, the `MessageConverter` mechanism will use the `contentType` header to parse the `String` into a `Vote` object.
+
+Just as with the other Spring Messaging methods, method arguments can be annotated with `@Payload`, `@Headers` and `@Header`. For methods that return data, `@SendTo` must be used for specifying the output binding destination for data returned by the methods as follows.
+
+[source,java]
+----
+@EnableBinding(Processor.class)
+public class TransformProcessor {
+
+  @Autowired
+  VotingService votingService;
+
+  @StreamListener(Processor.INPUT)
+  @SendTo(Processor.OUTPUT)
+	public VoteResult handle(Vote vote) {
+		return votingService.record(vote);
+	}
+}
+----
+
+NOTE: Content type headers can be either set by external applications in the case of Rabbit MQ, and they are supported as part of an extended internal protocol by Spring Cloud Stream for any type of transport (even the ones that do not support headers normally, like Kafka).
+
+== Configuration options reference
+
+The way Spring Cloud Stream behaves can be configured in two main ways:
+
+Configuration options can be provided per-binding. They include: Spring Cloud Stream-specific properties that control the general behavior of the binding, as well as consumer and producer properties for connecting to the middleware. A number of common consumer and producer properties are available to all binders, and some binders can specify additional properties that they support.
```

Comment:
Is this 'default' properties?  It isn't clear what are the two ways, global default vs. binder specific overrides?  Enumerations 1) and 2) would help.